packages/@aws-cdk/aws-s3/lib/notifications-resource/notifications-resource-handler.ts

```diff
@@ -53,22 +57,12 @@ export class NotificationsResourceHandler extends Construct {
   /**
    * The role of the handler's lambda function.
    */
-  public readonly role: iam.Role;
+  public readonly role: iam.IRole;
 
-  constructor(scope: Construct, id: string) {
+  constructor(scope: Construct, id: string, props: NotificationsResourceHandlerProps = {}) {
     super(scope, id);
 
-    this.role = new iam.Role(this, 'Role', {
-      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
-      managedPolicies: [
-        iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
-      ],
-    });
-
-    this.role.addToPolicy(new iam.PolicyStatement({
-      actions: ['s3:PutBucketNotification'],
-      resources: ['*'],
-    }));
+    this.role = props.role ?? this.createRole();
```

Comment:
I understand you're trying to avoid attaching policies on an imported role, but you actually don't know if this role is used anywhere further out. 

For example:

https://github.com/aws/aws-cdk/blob/534babde886646b6cd01f8799e3b4ed6db221263/packages/%40aws-cdk/aws-s3/lib/notifications-resource/notifications-resource.ts#L110

Instead, leave the policy handling as is, if the `IRole` is imported with `immutable: false` (which it should if that's what the caller wants), all these policies operations wont do anything. 

https://github.com/aws/aws-cdk/blob/534babde886646b6cd01f8799e3b4ed6db221263/packages/%40aws-cdk/aws-iam/lib/role.ts#L270-L272

Reply:
Thanks! I didn't know about `immutable: false`.